### PR TITLE
Take fix from r-dbi to allow correct retrieval of datetimeoffset

### PR DESCRIPTION
**Pull request by mkaes**
This PR will change the API behaviour of datetimeoffset.
It is no longer possible to get those columns via the timestamp class.
Since the timestamp class has no notion of UTC offset, allowing the datetimeoffset as a unparsed strings seems more reasonable.

### CODE REVIEW

File: test/mssql_test.cpp [hunk 1]
```diff
@@ -879,19 +878,17 @@ TEST_CASE_METHOD(mssql_fixture, "test_datetimeoffset", "[mssql][datetime]")
...
+		// the result is this NANODBC_TEXT("2006-12-30 13:45:12.3450000 -08:00");
+        REQUIRE(t.size() >= 27); // frac of seconds is server and system dependend
+        REQUIRE(t.substr(0, 23) == NANODBC_TEXT("2006-12-30 13:45:12.345"));
+		auto it = t.rbegin();
+		REQUIRE(*it++ == '0');
+		REQUIRE(*it++ == '0');
+		REQUIRE(*it++ == ':');
+		REQUIRE(*it++ == '8');
+		REQUIRE(*it++ == '0');
+		REQUIRE(*it++ == '-');
```
Comment (mloskot) [inline test/mssql_test.cpp:891]: The indentation here needs some correction here.

**Review by mloskot — CHANGES_REQUESTED** [review]
Thank you very much for the patch.

It looks good, just an indentation nitpick.

Could you also add to the PR description references to related r-dbi PR(s) ?

**Comment by mloskot** [comment]
@mkaes Can this one be also closed by https://github.com/nanodbc/nanodbc/pull/272 ?

File: test/mssql_test.cpp (hunk 1 above)
Comment (lexicalunit) [reply test/mssql_test.cpp:891]: Yep, looks like a mix of tabs and spaces.

Comment (mloskot) [reply test/mssql_test.cpp:891]: Interestingly, the ling CI job seems fine

Comment (lexicalunit) [inline test/mssql_test.cpp:891]:
```suggestion
        auto it = t.rbegin();
        REQUIRE(*it++ == '0');
        REQUIRE(*it++ == '0');
        REQUIRE(*it++ == ':');
        REQUIRE(*it++ == '8');
        REQUIRE(*it++ == '0');
        REQUIRE(*it++ == '-');
```

**Review by lexicalunit — CHANGES_REQUESTED** [review]

File: test/mssql_test.cpp (hunk 1 above)
Comment (lexicalunit) [reply test/mssql_test.cpp:891]: Hrm, linter still passing after rebase. Weird.

**Comment by lexicalunit** [comment]
Updated the style by running `find . \( -iname "*.cpp" -or -iname "*.h" \) -print0 | xargs -0 clang-format -i -style=file`

**Pull request merged**